### PR TITLE
🔙 `./db` backup / restore

### LINKED ISSUE #43: Add db backup shortcut
### Because

As a developer
I can run a single command to download a postgres dump to my local machine
because I need a quick and reliable way to do database backups.

### Done when

- [x] command downloads pgdump to local machine

### LINKED ISSUE #44: Add db restore shortcut
### Because

As a developer
I can run a single command to restore the database from a local postgres dump file
by specifying the arguments:
* path to the pg dump file
* which kube context i want to use (which corresponds to Demo or Production namespace)
because I need db restorations to be simple, fast, and reliable.

### Done when

- [x] There is a single command with minimal arguments that restores the database to either Prod or Demo namespaces.

### PR DESCRIPTION
**Pull request by mrharpo**
# `./db`
Adds `db` management CLI using [click](https://palletsprojects.com/p/click/)

## Commands
- `backup`
  - Creates a backup and save to a local file with the current timestamp
- `restore FILENAME`
  - Restores the database from a local SQL backup file

~~- `get-pod-name WORKLOAD`~~
~~- Gets the name of the first pod running the specified workload~~
~~- e.g. `get-pod-name db` -> `db-xyz123`~~

Having a hard time exposing `get-pod-name` as a CLI command. Click changes the call signature of the function, so we would need a wrapper for functions to use it internally. Removing from CLI as it is not strictly necessary.

## Options
- `-c` / `--context` (required)
  - `openvault` or `openvault-demo`

## Help
```
Usage: db [OPTIONS] COMMAND [ARGS]...

  Database CLI

  Management of Open Vault production databases

Options:
  -c, --context [openvault|openvault-demo]
                                  the kubectl context to use  [required]
  --help                          Show this message and exit.

Commands:
  backup        Backup the database to a local file
  restore       Restore the database from a local SQL backup file
```

Closes #43 
Closes #44

### CODE REVIEW

File: db [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    Management of Open Vault production databases
+    """
+    run(f'kubectl config use-context { context }')
+    click.echo(f'context set to { context }')
+
+
+@cli.command()
+@click.pass_context
+def backup(ctx):
```
Comment (afred) [inline db:29]: where is `ctx` coming from here?

Comment (mrharpo) [reply db:29]: @afred It's coming from the `@click.pass_context` decorator, from click: https://click.palletsprojects.com/en/8.1.x/commands/#nested-handling-and-contexts

Comment (afred) [reply db:29]: 👍

**Review by afred — APPROVED** [review]
👍

**Comment by mrharpo** [comment]
Closing, because we are going with an RDS database, which has built in backup / restore functionality.

**Pull request closed without merging**